src/isar/services/service_connections/mqtt/mqtt_service.py

```diff
@@ -0,0 +1,177 @@
+        self.mission_status_topic = config.get("mqtt_topics", "mission_status_topic")
+        self.mission_in_progress_topic = config.get(
+            "mqtt_topics", "mission_in_progress_topic"
+        )
+        self.current_mission_instance_id_topic = config.get(
+            "mqtt_topics", "current_mission_instance_id_topic"
+        )
+        self.current_mission_step_topic = config.get(
+            "mqtt_topics", "current_mission_step_topic"
+        )
+        self.mission_schedule_topic = config.get(
+            "mqtt_topics", "mission_schedule_topic"
+        )
+        self.current_state_topic = config.get("mqtt_topics", "current_state_topic")
+
+        self.start_mission_topic = config.get("mqtt_topics", "start_mission_topic")
+        self.start_mission_ack_topic = config.get(
+            "mqtt_topics", "start_mission_ack_topic"
+        )
+
+        self.stop_mission_topic = config.get("mqtt_topics", "stop_mission_topic")
+        self.stop_mission_ack_topic = config.get(
+            "mqtt_topics", "stop_mission_ack_topic"
+        )
+
+        self.connection_timeout = config.getfloat("mqtt", "connection_timeout")
+        self.connection_sleep = config.getfloat("mqtt", "connection_sleep")
+
+        self.host = host
+        self.port = port
+        self.client_id = client_id
+
+        self.mqtt_client = mqtt.Client(client_id=self.client_id)
+        self.mqtt_client.username_pw_set(username=username, password=password)
+
+        self.mqtt_client.on_connect = self._on_connect
+        self.mqtt_client.on_disconnect = self._on_disconnect
+
+        self.disconnection_time = None
+
+        self._connect()
+
+    def _connect(self):
+        try:
+            self.mqtt_client.connect(host=self.host, port=self.port, keepalive=60)
+        except ConnectionRefusedError:
+            self.logger.error("Error connecting to mqtt_broker")
+
+        self.mqtt_client.loop_start()
+        start_time = time.time()
+        while not self.mqtt_client.is_connected():
+            if time.time() - start_time > self.connection_timeout:
+                raise MQTTConnectionError
+            time.sleep(self.connection_sleep)
+
+    def _on_connect(self, mqttc, obj, flags, rc):
+        self.logger.info("Connected to MQTT-broker with result code: " + str(rc))
+        self.disconnection_time = None
+
+    def _on_disconnect(self, client, userdata, rc):
+        self.logger.info("Disconnected from MQTT-broker with result code: " + str(rc))
+        if not self.disconnection_time:
+            self.disconnection_time = time.time()
+
+    def is_connected(self):
+        return self.mqtt_client.is_connected()
+
+    def time_since_disconnect(self):
+        if not self.disconnection_time:
+            return 0
+        return time.time() - self.disconnection_time
+
+    def publish_status_message(self, status: Status):
+
+        self.mqtt_client.publish(
+            topic=self.mission_status_topic, payload=status.mission_status
```

Comment:
Note that the `Status` object should be removed with the implementation of MQTT. The object only existing to have a common structure to send through one queue. However, now we are sending across many topics and the entries within `Status` should be in a flat structure in `StateMachine`.